### PR TITLE
feat: light memoize decorator

### PR DESCRIPTION
**Pull request by ala-n**
Update memoize decorator inner logic to use simple value caching for members simple property so 

### current strategy is
 - Static method -> replace to memo-decorated method
 - Static getter -> replace getter function to memo-decorated
 
 - object method -> memo-decorated own property method
 - object getter -> own property computed value _(**new**)_

### CODE REVIEW

File: src/modules/esl-utils/decorators/memoize.ts [hunk 1]
```diff
@@ -47,9 +61,9 @@ function memoizeMember(originalMethod: any, prop: string, isGetter: boolean, has
  * Accepts not own properties.
```
Comment (ala-n) [inline src/modules/esl-utils/decorators/memoize.ts:61]: TODO: add warning to doc

**Review by julia-murashko — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

**Review by andreybelous — APPROVED** [review]

**Review by dshovchko — APPROVED** [review]

File: src/modules/esl-utils/decorators/memoize.ts [hunk 2]
```diff
@@ -12,32 +12,46 @@ export function memoize<H extends MemoHashFn>(hashFn: H): MethodTypedDecorator<(
...
 
 // Lock storage to prevent cache logic for some key
 const locks = new WeakMap();
+const defineOwnKeySafe = (obj: any, prop: string, value: any) => {
+  locks.set(obj, prop); // IE try to get key with the prototype instance call, so we lock it
+  Object.defineProperty(obj, prop, {value, writable: true, configurable: true});
+  locks.delete(obj); // Free property key
+};
+
+/** Cache getter result as an object ovn property */
```
Comment (nattallius) [inline src/modules/esl-utils/decorators/memoize.ts:39]: own

**Review by nattallius — APPROVED** [review]

File: src/modules/esl-utils/decorators/memoize.ts (hunk 2 above)
Comment (ala-n) [reply src/modules/esl-utils/decorators/memoize.ts]: Thanks! Fixed.

**Review by abarmina — APPROVED** [review]

**Pull request merged**